spec/features/publish/managing_users_spec.rb

```diff
@@ -151,11 +266,51 @@ def and_i_click_remove_user
     users_show_page.remove_user_link.click
   end
 
+  def given_i_click_change_email_address
+    users_show_page.change_email_address.click
+  end
+
+  def given_i_click_change_first_name
+    users_show_page.change_first_name.click
+  end
+
+  def given_i_click_change_last_name
+    users_show_page.change_last_name.click
+  end
+
   def and_i_confirm
     users_delete_page.remove_user_button.click
   end
 
   def then_the_user_should_be_deleted
     expect(provider_users_page).not_to have_text "Mr Cool"
   end
+
+  def and_the_new_email_is_displayed
+    expect(page).to have_text("a-changed-email@address.com")
+  end
+
+  def and_i_click_update_user
+    click_button "Update user"
+  end
+
+  def then_i_should_see_a_validation_error_message
+    expect(page).to have_text("Enter an email address in the correct format, like name@example.com")
+  end
+
+  def then_the_first_name_should_have_updated_in_the_database
+    expect(Provider.find_by(provider_name: "Batman's Chocolate School").users.exists?(first_name: "New first name")).to be(true)
```

Comment:
can we put `Provider.find_by(provider_name: "Batman's Chocolate School").users.exists?(first_name: "New first name")` into a descriptive method which can be used in these assertions?

Reply:
Something like this? https://github.com/DFE-Digital/publish-teacher-training/pull/3185/commits/059d2b994be3822b83ccd432060f0897a2d9b672

Reply:
Done: https://github.com/DFE-Digital/publish-teacher-training/pull/3185/commits/cb6a4bb90982e930a4ff20cff315bf1944e451e6